### PR TITLE
ci: pin `actions/checkout` to a commit

### PR DESCRIPTION
**Pull request by G-Rath**
I noticed that this hasn't been pinned, which it should be to make scorecard happier

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Check out code
-        uses: actions/checkout@v3
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```
Comment (G-Rath) [inline .github/workflows/lint.yaml:33]:
```suggestion
        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```

**Comment by G-Rath** [comment]
(there's only one more of these to do I think 😅)

**Review by cuixq — APPROVED** [review]
I think it worth to mention v3 upgraded to v4 in the PR description.

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -39,7 +39,7 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v3
+      uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```
Comment (G-Rath) [inline .github/workflows/codeql-analysis.yml:42]:
```suggestion
      uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
```

File: .github/workflows/lint.yaml [hunk 3]
```diff
@@ -30,7 +30,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Check out code
-        uses: actions/checkout@v3
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```
Comment (G-Rath) [inline .github/workflows/lint.yaml:33]:
```suggestion
      uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
```

**Comment by G-Rath** [comment]
> I think it worth to mention v3 upgraded to v4 in the PR description.

I've just reverted that, since renovate should end up doing it or otherwise I can do it in a dedicated PR later

**Comment by another-rex** [comment]
I'm not sure if renovate is configured to do major version updates for workflows.

**Pull request merged**